### PR TITLE
Dictionaries to dataclass

### PR DESCRIPTION
**Pull request by khalford**
Created the device dataclass. Created 2 methods to take the csv file and create a dataclass object for each class.

### CODE REVIEW

**Review by Protabs — APPROVED** [review]

File: pynetbox_data_uploader/lib/utils/csv_to_dataclass.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            status=row["status"],
+            site=row["site"],
+            location=row["location"],
+            rack=row["rack"],
+            face=row["face"],
+            airflow=row["airflow"],
+            position=row["position"],
+            name=row["name"],
+            serial=row["serial"],
+        )
```
Comment (anish-mudaraddi) [inline pynetbox_data_uploader/lib/utils/csv_to_dataclass.py:39]: This is fine - but you could create a method called `from_dict` in the dataclass to make this a bit tidier 
```
from dataclasses import dataclass, fields

from_dict(device_dict: Dict):
        field_set = {field.name for field in fields(Device) if field.init}
        filtered_arg_dict = {
            key: value for key, value in device_dict.items() if key in field_set
        }
        return Device(**filtered_arg_dict)
```

File: pynetbox_data_uploader/lib/utils/dataclass_data.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    device_type: str
+    status: str
+    site: str
+    location: str
+    rack: str
+    face: str
+    airflow: str
+    position: str
+    name: str
+    serial: str
```
Comment (anish-mudaraddi) [inline pynetbox_data_uploader/lib/utils/dataclass_data.py:23]: Are any of these attribute optional? Can we set some of these to a default value?

File: pynetbox_data_uploader/lib/utils/csv_to_dataclass.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+import csv
+from typing import List
+from lib.utils.dataclass_data import Device
+
+
+def open_file(file_path: str) -> csv.DictReader:
```
Comment (anish-mudaraddi) [inline pynetbox_data_uploader/lib/utils/csv_to_dataclass.py:6]: Whenever you're using file paths - you should use Pathlib to validate the filepath before you use it 

https://docs.python.org/3/library/pathlib.html

**Review by anish-mudaraddi — CHANGES_REQUESTED** [review]
looks good - I made a few points - much of it can be ignored - just something to think about

File: pynetbox_data_uploader/.pylintrc [hunk 4]
```diff
@@ -5,6 +5,5 @@ max-line-length=120
 
 # Disable various warnings:
 # C0114: Missing module string - we don't need module strings for the small repo
-# W0511: TODOs they're well....to do later
 
-disable=C0114,W0511
```
[the remark this answers is not in the document]
Comment (khalford) [reply pynetbox_data_uploader/.pylintrc:10]: A David F nitpick. It hasn't been used anywhere in the code, so doesn't really need to be there.

File: pynetbox_data_uploader/lib/utils/dataclass_data.py (hunk 2 above)
Comment (khalford) [reply pynetbox_data_uploader/lib/utils/dataclass_data.py:23]: Some are optional, none have a default value.

File: pynetbox_data_uploader/lib/utils/csv_to_dataclass.py (hunk 1 above)
Comment (khalford) [reply pynetbox_data_uploader/lib/utils/csv_to_dataclass.py:39]: Will consider that in future PR.

File: pynetbox_data_uploader/lib/utils/dataclass_data.py (hunk 2 above)
Comment (anish-mudaraddi) [reply pynetbox_data_uploader/lib/utils/dataclass_data.py]: if some are optional the default value should be None - you'll get an error if the user doesn't provide these. Set these values to None in the dataclass

File: pynetbox_data_uploader/lib/utils/dataclass_data.py [hunk 5]
```diff
@@ -16,8 +16,8 @@ class Device:
     site: str
     location: str
     rack: str
-    face: str
-    airflow: str
+    face: Optional[str]
+    airflow: Optional[str]
```
Comment (anish-mudaraddi) [inline pynetbox_data_uploader/lib/utils/dataclass_data.py:20]: if these are optional then you need to provide a default value of None

File: pynetbox_data_uploader/lib/user_methods/csv_to_netbox.py [hunk 6]
```diff
@@ -35,6 +36,10 @@ def read_csv(self, file_path) -> List:
         :return: Returns a list of devices
         """
         print("Reading CSV...")
+        try:
+            Path(file_path).resolve()
+        except (OSError, RuntimeError):
+            raise Exception("The given path is not valid.")
```
Comment (anish-mudaraddi) [inline pynetbox_data_uploader/lib/user_methods/csv_to_netbox.py:42]: did you add a test case for this? You can mock Path

**Review by Protabs — APPROVED** [review]

**Review by anish-mudaraddi — APPROVED** [review]

**Pull request merged**